remoting/src/main/java/org/apache/rocketmq/remoting/netty/NettyRemotingAbstract.java

```diff
@@ -355,19 +355,22 @@ private Runnable buildProcessRequestHandler(ChannelHandlerContext ctx, RemotingC
     public void processResponseCommand(ChannelHandlerContext ctx, RemotingCommand cmd) {
         final int opaque = cmd.getOpaque();
         final ResponseFuture responseFuture = responseTable.get(opaque);
+        final String remoteAddr = ctx == null ? "" : RemotingHelper.parseChannelRemoteAddr(ctx.channel());
         if (responseFuture != null) {
             responseFuture.setResponseCommand(cmd);
 
             responseTable.remove(opaque);
 
+            doAfterRpcHooks(remoteAddr, responseFuture.getRequestCommand(), cmd);
```

Comment:
maybe invokesync will call this method twice?

Reply:
I mean in func invokesync of NettyRemotingClient, the doAfterRpcHooks is called now, if you call doAfterRpcHooks in this place, may the doAfterRpcHooks will be called twice?

Reply:
u r right,  I didn't consider that aspect. So we should delete the doAfterRpcHooks in func invokeSync.